src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
Useless parameterization, you probably want to parameterize this interface so that it can be parameterized when implemented only by the descendant class Machine, but that's just my opinion
```suggestion
public interface MachineProducer<T extends Machine> {
```